<a href="https://colab.research.google.com/github/Christinathattil/MACHINE-LEARNING-PROJECTS/blob/Main/2348511_CIA2_TEST.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [52]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [54]:
df=pd.read_csv('/content/2015.csv')
df.head()

Country          Region  Happiness Rank  Happiness Score  \
0  Switzerland  Western Europe               1            7.587   
1      Iceland  Western Europe               2            7.561   
2      Denmark  Western Europe               3            7.527   
3       Norway  Western Europe               4            7.522   
4       Canada   North America               5            7.427   

   Standard Error  Economy (GDP per Capita)   Family  \
0         0.03411                   1.39651  1.34951   
1         0.04884                   1.30232  1.40223   
2         0.03328                   1.32548  1.36058   
3         0.03880                   1.45900  1.33095   
4         0.03553                   1.32629  1.32261   

   Health (Life Expectancy)  Freedom  Trust (Government Corruption)  \
0                   0.94143  0.66557                        0.41978   
1                   0.94784  0.62877                        0.14145   
2                   0.87464  0.64938                        0.48357   
3                   0.88521  0.66973                        0.36503   
4                   0.90563  0.63297                        0.32957   

   Generosity  Dystopia Residual  
0     0.29678            2.51738  
1     0.43630            2.70201  
2     0.34139            2.49204  
3     0.34699            2.46531  
4     0.45811            2.45176

In [55]:
df.shape

(158, 12)

In [56]:
df.isnull().sum()

Country                          0
Region                           0
Happiness Rank                   0
Happiness Score                  0
Standard Error                   0
Economy (GDP per Capita)         0
Family                           0
Health (Life Expectancy)         0
Freedom                          0
Trust (Government Corruption)    0
Generosity                       0
Dystopia Residual                0
dtype: int64

No missing values

In [61]:
bins_values=[3,5,7,8]
bins_labels=['Low Happiness','Medium Happiness','High Happiness']

df=pd.cut(df['Happiness Score'], bins_values,bins_labels)
df

KeyError: ignored

In [65]:
encoded_df=pd.getdummies(df['Region'],axis=1)

AttributeError: ignored

In [67]:
from scipy import stats
stats.zscore(df['Health(Life Expectancy)'])

KeyError: ignored

In [68]:
from sklearn.preprocessing import MinMaxScaler()

SyntaxError: ignored

In [69]:
from scipy import chi2contigency

chi2,p_value,dof

if p_value<0.05:
  print("The null hypothesis is accepted i.e they are independent of each other")

if p_value>0.05:
  print("The null hypothesis is rejected i.e they are related of each other")

SyntaxError: ignored

In [ ]:
'''A stacked bar chart showing the distribution of Region,Happiness
Score group'''
values=df['Happiness Score']
plt.scatter(df, y=values, label='Happiness Score')


In [ ]:
cov_df=df['Happiness Score'].cov
cov_df

In [ ]:
sns.heatmp(cov_df)

In [ ]:
sns.boxplot(cov_df , y: df['Region'], hue: green )

After the Exploratory data analysis, we can come to the conclusion as:

1.There are no null values
2. 158 rows and 12 columns are present
3. Binning the Happiness score helps to categorize data as well as convert categorical data to discrete data
4. One hot encoding is performed on categorize attributes by converting it into binary matrix
5. Z score normalisation is used if there are more outliers int the dataset to normalise them between -1 and +1.If the values are more than equal to 2, then there is a relationship between both the attributes.

6. Chi square test is performed to check whether there is a significant relationship between Region and Happiness Rank columns ie., categorical values.If the p value is less than the significant value, then the null hypothesis is rejected. else both are related to each other.

7. Covariance matrix is used to identify the degree of dispersion between the attributes.Correlation matrix helps to find the quantitative value of dispersion between the attributes.

8. The heatmap represents the Covariance matrix in a visualised form. However the diagonal of the heatmap will always be 1 since those values are compared to themselves.